src/helper/helper.js

```diff
@@ -48,165 +48,163 @@ goog.module('helper');
 
 const {type, hasOwn, isPlainObject} = goog.require('plain');
 
-
 /**
- * Creates a new helper object for the given dataLayer.
+ * A helper that will listen for new messages on the given dataLayer.
+ * Each new message will be merged into the helper's "abstract data model".
+ * This internal model object holds the most recent value for all keys which
+ * have been set on messages processed by the helper.
  *
- * @constructor
- * @param {!Array.<!Object>} dataLayer The dataLayer to help with.
- * @param {function(!Object, !Object)=} opt_listener The callback function to
- *     execute when a new state gets pushed onto the dataLayer.
- * @param {boolean=} opt_listenToPast If true, the given listener will be
- *     executed for state changes that have already happened.
+ * You can retrieve values from the data model by using the helper's get method.
  */
-const DataLayerHelper = function(dataLayer, opt_listener, opt_listenToPast) {
-
+class DataLayerHelper {
   /**
-   * The dataLayer to help with.
-   * @type {!Array.<!Object>}
-   * @private
-   */
-  this.dataLayer_ = dataLayer;
-
-  /**
-   * The listener to notify of changes to the dataLayer.
-   * @type {function(!Object, !Object)}
-   * @private
+   * Creates a new helper object for the given dataLayer.
+   *
```

Comment:
nit: Not a huge deal, but please be consistent with the empty JSDoc lines between the description and the parameter annotations (either do it for all or do it for none).

Reply:
I believe all methods have a space between the description and @param. The JSDoc comments were styled like this before we got to it and I just (hopefully) followed the convention elsewhere too.

Reply:
Gotcha. Thanks!